cli/src/commands/validate/validate.spec.ts

```diff
@@ -0,0 +1,180 @@
+        fetchMock.restore();
+    });
+
+    it('The pattern instantiation file at given URL returns non JSON response', async () => {
+        const apiGateway = readFileSync(path.resolve(__dirname, '../../../test_fixtures/api-gateway.json'), 'utf8');
+
+        const markdown = ' #This is markdown';
+        fetchMock.mock('http://exist/api-gateway.json', apiGateway);
+        fetchMock.mock('https://url/with/non/json/response', markdown);
+
+        await expect(validate('https://url/with/non/json/response', 'http://exist/api-gateway.json', metaSchemaLocation))
+            .rejects
+            .toThrow();
+
+        expect(mockExit).toHaveBeenCalledWith(1);
+        fetchMock.restore();
+    });
+
+
+    it('The pattern instantiation does not match the json schema', async () => {
+        const apiGateway = readFileSync(path.resolve(__dirname, '../../../test_fixtures/api-gateway.json'), 'utf8');
+        
+
+        fetchMock.mock('http://exist/api-gateway.json', apiGateway);
+        const apiGatewayInstantiation = readFileSync(path.resolve(__dirname, '../../../test_fixtures/api-gateway-implementation-that-does-not-match-schema.json'), 'utf8');
+        fetchMock.mock('https://exist/api-gateway-implementation.json', apiGatewayInstantiation);
+
+        await expect(validate('https://exist/api-gateway-implementation.json', 'http://exist/api-gateway.json', metaSchemaLocation))
+            .rejects
+            .toThrow();
+
+        expect(mockExit).toHaveBeenCalledWith(1);
+        fetchMock.restore();
+    });
+
+    it('The pattern instantiation does not pass all the spectral rules', async () => {
+
+        const expectedSpectralOutput: ISpectralDiagnostic[] = [
+            {
+                code: 'no-empty-properties',
+                message: 'Must not contain string properties set to the empty string or numerical properties set to zero',
+                severity: 0,
+                path: JSON.parse(JSON.stringify('$..*')),
+                range: { start: { line: 1, character: 1 }, end: { line: 2, character: 1 } }
+            }
+        ];
+
+        mockRunFunction.mockReturnValue(expectedSpectralOutput);
+
+        const apiGateway = readFileSync(path.resolve(__dirname, '../../../test_fixtures/api-gateway.json'), 'utf8');
+        fetchMock.mock('http://exist/api-gateway.json', apiGateway);
+
+        const apiGatewayInstantiation = readFileSync(path.resolve(__dirname, '../../../test_fixtures/api-gateway-implementation-that-does-not-pass-the-spectral-validation.json'), 'utf8');
+        fetchMock.mock('https://exist/api-gateway-implementation.json', apiGatewayInstantiation);
+
+        await expect(validate('https://exist/api-gateway-implementation.json', 'http://exist/api-gateway.json', metaSchemaLocation))
+            .rejects
+            .toThrow();
+
+        expect(mockExit).toHaveBeenCalledWith(1);
+        fetchMock.restore();
+    });
+
+    it('The meta schema location is not a directory', async () => {
```

Comment:
best practice for test names with jest is to write it as if it would come after the word in the `describe()` function so that the output reads like a sentence. E.g. we could name the test `exits with error when meta schema location is not a directory`, so that the test output reads as `validate exits with error... etc.`